csharp/src/Google.Protobuf/Collections/RepeatedField.cs

```diff
@@ -101,10 +101,11 @@ public void AddEntriesFrom(CodedInputStream input, FieldCodec<T> codec)
                 int length = input.ReadLength();
                 if (length > 0)
                 {
+                    EnsureSize(count + length);
```

Comment:
I know nothing about C# partial message, but the ability to customize parsing/serializing routines is what I would be afraid to offer. Such customization comes with a close tie with the implementation details and makes it hard or impossible for us to make changes later. It also causes interoperability problems because other implementations (official implementations in other languages and third-party ones) will not be able to recognize such payload bytes with custom content.